app/controllers/agents_controller.rb

```diff
@@ -179,7 +179,9 @@ def complete
 
   # Sanitize params[:return] to prevent open redirect attacks, a common security issue.
   def redirect_back(message)
-    if params[:return] == "show" && @agent
+    if params[:return] == "show" && @agent.destroyed?
```

Comment:
It would be saver to check if the `@agent` variable is present, there might be use cases of this helper where it is not the case.